packages/server-wallet/src/engine/index.ts

```diff
@@ -12,16 +16,29 @@ import {SingleThreadedEngine} from './engine';
  */
 export abstract class Engine extends SingleThreadedEngine implements EngineInterface {
   static async create(
-    engineConfig: IncomingEngineConfig
+    engineConfig: EngineConfig,
+    logger: P.Logger
```

Comment:
> Why is it that the Engine now requires a logger? I see that it's being passed to the Store, which optionally takes it.
>
> Is the take-home just that the logger provides for better feedback from the Store on testing?
> 
> Should the logger be optional for the Engine as well?

Right now we log out a lot of information to the logger so I think using a `logger` should be opt-out instead of opt-in. Since we're always creating a[ logger in the wallet](https://github.com/statechannels/statechannels/blob/ag/config-clean-up/packages/server-wallet/src/wallet/wallet.ts#L68:L68) I think it makes sense to require it in the engine and all the sub-components. (which means we should probably update the `store` to require it as well)

Reply:
This seems reasonable to me!